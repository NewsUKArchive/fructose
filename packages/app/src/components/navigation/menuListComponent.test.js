/* globals describe it expect */
import React from "react";
import { configure, shallow } from "enzyme";
import Adapter from "enzyme-adapter-react-16";

import MenuListComponent from "./menuListComponent";

configure({ adapter: new Adapter() });

describe("Menu List Component", () => {
  const expectedMenuHeader = "Test Header";
  const expectedMenuItems = ["item1", "item2"];
  const expectedOnMenuItemPress = jest.fn();
  const componentUnderTest = (
    <MenuListComponent
      menuHeader={expectedMenuHeader}
      menuItems={expectedMenuItems}
      onMenuItemPress={expectedOnMenuItemPress}
    />
  );

  it("renders specified component", () => {
    const wrapper = shallow(componentUnderTest);
    expect(wrapper).toMatchSnapshot();
  });
});
