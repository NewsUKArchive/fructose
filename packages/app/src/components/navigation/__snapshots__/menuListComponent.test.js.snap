// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Menu List Component renders specified component 1`] = `
<FlatList
  ItemSeparatorComponent={[Function]}
  ListHeaderComponent={
    <Text
      accessible={true}
      allowFontScaling={true}
      ellipsizeMode="tail"
      style={
        Object {
          "color": "black",
          "fontSize": 24,
          "margin": 4,
          "marginBottom": 8,
        }
      }
    >
      Test Header
    </Text>
  }
  data={
    Array [
      Object {
        "key": "item1",
      },
      Object {
        "key": "item2",
      },
    ]
  }
  disableVirtualization={false}
  horizontal={false}
  initialNumToRender={10}
  keyExtractor={[Function]}
  maxToRenderPerBatch={10}
  numColumns={1}
  onEndReachedThreshold={2}
  renderItem={[Function]}
  scrollEventThrottle={50}
  updateCellsBatchingPeriod={50}
  windowSize={21}
/>
`;
